src/components/LightBox/LightBox.tsx

```diff
@@ -0,0 +1,245 @@
+import { ExitFullscreenIcon } from '../Icons/ExitFullscreenIcon'
+import { FullscreenIcon } from '../Icons/FullscreenIcon'
+import { HEADER_HEIGHT_PX } from '../NavBar/NavBar'
+
+type UseLightBoxReturnType = {
+  getStyle: (element: HTMLElement | null) => React.CSSProperties
+  close: () => void
+  display: (element: HTMLElement) => void
+  isDisplayedElement: (el: HTMLElement) => boolean
+  isActive: boolean
+  isMobile: boolean
+}
+
+// Main logic was taken from:
+// https://github.com/acid-info/logos-docusaurus-plugins/blob/964bf92263cf6eb61527c92c83c73ab6ba74e36d/packages/logos-docusaurus-theme/src/client/containers/LightBox/LightBox.tsx
+export const useLightBox = (): UseLightBoxReturnType => {
+  const { y: scrollY } = useWindowScroll()
+  const [displayedElement, setDisplayedElement] = useState<HTMLElement | null>(
+    null,
+  )
+  const [displayedStyle, setDisplayedStyle] = useState<CSSProperties>({
+    opacity: '0.5',
+  })
+  const isMobile = useIsMobile()
+
+  const defaultStyle: CSSProperties = {
+    opacity: 1,
+    transform: 'scale(1) translate(0px, 0px)',
+    transition: 'all 0.3s ease-in-out',
+  }
+
+  const display = (element: HTMLElement) => {
+    setDisplayedElement(element)
+
+    const vw = document.body.clientWidth
+    const vh = window.innerHeight
+
+    const maxWidth = isMobile ? vw - 32 : vw * 0.9375
+    const maxHeight = vh - 128
+
+    const rect = element.getBoundingClientRect()
+
+    const scale = Math.min(maxHeight / rect.height, maxWidth / rect.width)
+
+    const center = [rect.left + rect.width / 2, rect.top + rect.height / 2]
+    const windowCenter = [vw / 2, vh / 2]
+
+    const translate = windowCenter.map((w, i) => (w - center[i]!) / scale)
+
+    setDisplayedStyle({
+      zIndex: 202,
+      transform: `scale(${scale}) translate(${translate[0]}px, ${translate[1]}px)`,
+      position: 'relative',
+    })
+  }
+
+  const close = () => {
+    setDisplayedElement(null)
+  }
+
+  // If the user scrolls, close the lightbox.
+  useEffect(() => {
+    // On mobile, this is not needed: even if the user attempts to scroll, lightbox stays open.
+    if (isMobile) return
+
+    if (displayedElement) close()
+  }, [scrollY])
+
+  // Only for mobile - toggle the whole page's overflow when the lightbox is displayed.
+  useEffect(() => {
+    const html = document.querySelector('html')!
+    html.style.overflow = isMobile && displayedElement ? 'hidden' : 'initial'
+  }, [isMobile, displayedElement])
+
+  return {
+    getStyle: (element: HTMLElement | null) => ({
+      ...defaultStyle,
+      ...(element === displayedElement ? displayedStyle : {}),
+    }),
+    close,
+    display,
+    isDisplayedElement: (el: HTMLElement) => displayedElement === el,
+    isActive: !!displayedElement,
+    isMobile,
+  }
+}
+
+type OnUpdateParams = {
+  x: number
+  y: number
+  scale: number
+}
+
+type LightBoxProps = {
+  children: React.ReactNode
+}
+
+export const LightBox = ({ children }: LightBoxProps) => {
```

Comment:
Created a new LightBox component here. Maybe in the future we should have a single place for this? Instead of having 2 implementations - 1 for LPE and one for vac.dev